channel/conclude.go

```diff
@@ -160,7 +165,21 @@ func (a *Adjudicator) checkConcludedState(
 	}
 }
 
-func (a *Adjudicator) waitConcludedSecondary(ctx context.Context, req channel.AdjudicatorReq, events chan *subscription.Event) (concluded bool, err error) {
+func (a *Adjudicator) waitConcludedSecondary(ctx context.Context, req channel.AdjudicatorReq) (concluded bool, err error) {
+	sub, err := subscription.Subscribe(ctx, a.ContractBackend, a.bound, updateEventType(req.Params.ID()), startBlockOffset, a.txFinalityDepth)
```

Comment:
What is the reason that we now create the event sub in the method itself than providing it as a parameter? 
Because now this code part is duplicated since we also create it again in L75.

Reply:
The method before was just very hard to understand, because the functionalities were not clearly separated. And yes, there is now some code duplication, which is not too great. I dedicated some more time to it and was able to remove a bunch of code duplication.